### ISSUE TITLE
Custom Format Bitrate Condition

### ISSUE THREAD

**Issue by clifford64** [Sonarr/Sonarr#5496]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

I recently upgraded to Sonarr v4 beta to start experimenting with Custom Formats. My current workflow is everything I get gets converted to HEVC if it isn't already in that format. I run this workflow because I care more about the space savings benefits of x265 that I do about the slight quality loss. I also believe most people run a workflow like this for similar reasons.

v4 helps with this process because now I can force my workflow to prefer HEVC from the beginning, if it doesn't find one, that's ok, it can get a normal x264 version and then it automatically gets converted. I have Sonarr setup to not force upgrades to HEVC, so even if an HEVC release comes up, I don't want to it to upgrade because it will have already been converted.

The problem I am currently running into is that some HEVC releases are fairly large and is not helping with the space savings functionality I am looking for. Custom Formats has a condition to specify file size, but I think that is too broad to apply that condition across 30, 45, 60+ runtimes. 

### **tldr**; 
Custom Formats can specify max file size as a condition, but applying this concept to a range of runtime shows isn't accurate enough. Just like quality definitions, a condition for bitrate or MB/GB per hour might be more useful and accurate.

### Describe the solution you'd like

Add or replace custom format condition for file size to be bitrate instead to make the custom format more accurate for the wider ranges of runtimes. 

### Describe alternatives you've considered

Making multiple custom formats is possible based on runtimes, but this would be tedious and get complicated fast by trying to manually adjust the profile for each item.

**Comment by markus101** [comment]
You can set preferred size (per quality) on Settings: Quality, which is based on run time. No plans to add preferred size to CF at this time.

**Comment by clifford64** [comment]
> You can set preferred size (per quality) on Settings: Quality, which is based on run time. No plans to add preferred size to CF at this time.

The problem with this method is preferred size is going to be different depending on format. Setting the same size with different formats will produce dramatically different qualities.

**Comment by markus101** [comment]
Similar issue here in that size as MB / minute like we do for qualities would apply to everything or you'd need multiple custom formats to properly account for different resolutions/sources.

https://github.com/Sonarr/Sonarr/issues/613 is would probably help the most, though it's never gotten much traction.

**Comment by clifford64** [comment]
> Similar issue here in that size as MB / minute like we do for qualities would apply to everything or you'd need multiple custom formats to properly account for different resolutions/sources.
> 
> #613 is would probably help the most, though it's never gotten much traction.

For someone with the workflow of wanting hevc or custom formats for similar quality, but smaller file size, that wouldn't be an issue because the size in MB/minute would be the same between different resolutions and bitrates. And to clarify, the idea of a bitrate condition on the custom format I think would best fit in with preferred bitrate or max bitrate rather than an absolute. That way, if a release is offered in HEVC and it fits within my range of wanting about 1GB per 45 minute file, then it will go for that. Otherwise, it gets the normal x264 file at 3GB or so and converts it to the 1GB or less through my workflow. 

The option for file size is already in custom formats, this idea is just building on that slightly to be a little more accurate for the formats wanted. Applying the idea of a Max file size of 2GB to all files is going to yield dramatically different results between 15 minute runtimes and 60 minute runtimes.
